components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1669,22 +1670,27 @@ void AdsServiceImpl::GetScheduledCaptcha(
 
 void AdsServiceImpl::ShowScheduledCaptchaNotification(
     const std::string& payment_id,
-    const std::string& captcha_id) {
+    const std::string& captcha_id,
+    const bool should_show_notification) {
 #if BUILDFLAG(BRAVE_ADAPTIVE_CAPTCHA_ENABLED)
   PrefService* pref_service = profile_->GetPrefs();
-  if (pref_service->GetBoolean(
-          brave_adaptive_captcha::kScheduledCaptchaPaused)) {
-    VLOG(0) << "Ads paused; support intervention required";
-    return;
-  }
+  if (should_show_notification) {
+    if (pref_service->GetBoolean(
+            brave_adaptive_captcha::prefs::kScheduledCaptchaPaused)) {
+      VLOG(0) << "Ads paused; support intervention required";
+      return;
+    }
 
-  const int snooze_count = pref_service->GetInteger(
-      brave_adaptive_captcha::kScheduledCaptchaSnoozeCount);
+    const int snooze_count = pref_service->GetInteger(
+        brave_adaptive_captcha::prefs::kScheduledCaptchaSnoozeCount);
 
-  ads_tooltips_delegate_->ShowCaptchaTooltip(
-      payment_id, captcha_id, snooze_count == 0,
-      base::BindOnce(&AdsServiceImpl::ShowScheduledCaptcha, AsWeakPtr()),
-      base::BindOnce(&AdsServiceImpl::SnoozeScheduledCaptcha, AsWeakPtr()));
+    ads_tooltips_delegate_->ShowCaptchaTooltip(
```

Comment:
`ads_tooltips_delegate_` can be `nullptr`?

Reply:
i won't be nullptr  for desktop as `should_show_notification` would be true in desktop always.

Reply:
Adding DCHECK now